src/Microsoft.ML.Onnx/OnnxUtils.cs

```diff
@@ -349,5 +352,77 @@ public ModelArgs(string name, TensorProto.Types.DataType dataType, List<long> di
 
             return new ModelArgs(name, dataType, dimsLocal, dimsParamLocal);
         }
+
+        // Make int64 scalar in ONNX from native C# number
+        public static TensorProto MakeInt64(string name, long value)
+        {
+            var tensor = new TensorProto();
+            tensor.Name = name;
+            tensor.DataType = TensorProto.Types.DataType.Int64;
+            tensor.Int64Data.Add(value);
+            return tensor;
+        }
+
+        // Make float vector (i.e., 1-D tensor) with dims=null. Otherwise, dims is used as the shape of the produced tensor.
+        public static TensorProto MakeInt64s(string name, IEnumerable<long> values, IEnumerable<long> dims = null)
+        {
+            var tensor = new TensorProto();
+            tensor.Name = name;
+            tensor.DataType = TensorProto.Types.DataType.Int64;
+            tensor.Int64Data.AddRange(values);
+            if (dims != null)
+                tensor.Dims.AddRange(dims);
+            else
+                tensor.Dims.Add(values.Count());
+            return tensor;
+        }
+
+        // Make float scalar in ONNX from native C# number
+        public static TensorProto MakeFloat(string name, float value)
+        {
+            var tensor = new TensorProto();
+            tensor.Name = name;
+            tensor.DataType = TensorProto.Types.DataType.Float;
+            tensor.FloatData.Add(value);
+            return tensor;
+        }
+
+        // Make float vector (i.e., 1-D tensor) with dims=null. Otherwise, dims is used as the shape of the produced tensor.
+        public static TensorProto MakeFloats(string name, IEnumerable<float> values, IEnumerable<long> dims = null)
+        {
+            var tensor = new TensorProto();
+            tensor.Name = name;
+            tensor.DataType = TensorProto.Types.DataType.Float;
+            tensor.FloatData.AddRange(values);
+            if (dims != null)
+                tensor.Dims.AddRange(dims);
+            else
+                tensor.Dims.Add(values.Count());
```

Comment:
tensor.Dims.AddRange(dims??values.Count());

Reply:
We can't do this because dims and values.Count() have different types.